File: problem/Longest_Balanced_Subarray_I/longestBalanced.py
class Solution:
    def longestBalanced(self, nums):
        n = len(nums)
        res = 0
        seen_arr = [0] * 100001
        leet = 1 
        
        for i in range(n):
            if n - i <= res:
                break
                
            counts = [0, 0]
            marker = (leet << 16) | (i + 1)
            
            for j in range(i, n):
                val = nums[j]
                if seen_arr[val] != marker:
                    seen_arr[val] = marker
                    counts[val & 1] += 1
                
                if counts[0] == counts[1]:
                    res = max(res, j - i + 1)
        
        return res

File: problem/Longest_Balanced_Subarray_I/longestBalanced.c
#include <stdint.h>

#define MAX_VAL 100001
uint32_t seen[MAX_VAL] = {0};
uint32_t leet = 0;

int longestBalanced(int *nums, int numsSize)
{
    leet++;
    int res = 0;

    for (int i = 0; i < numsSize && numsSize - i > res; i++)
    {
        int A[2] = {0, 0};
        uint32_t marker = (leet << 16) | (uint32_t)(i + 1);

        for (int j = i; j < numsSize; j++)
        {
            int val = nums[j];
            if (seen[val] != marker)
            {
                seen[val] = marker;
                A[val & 1]++;
            }
            if (A[0] == A[1])
            {
                int len = j - i + 1;
                if (len > res)
                    res = len;
            }
        }
    }

    return res;
}